addons/account_peppol/models/res_company.py

```diff
@@ -268,3 +271,21 @@ def _get_peppol_edi_mode(self):
         # by design, we can only have zero or one proxy user per company with type Peppol
         peppol_user = self.sudo().account_edi_proxy_client_ids.filtered(lambda u: u.proxy_type == 'peppol')
         return peppol_user.edi_mode or config_param or 'prod'
+
+    def _get_company_info_on_peppol(self, edi_identification):
+        self.ensure_one()
+        is_company_on_peppol = False
+        external_provider = None
+        error_msg = ''
+        if (
+            (participant_info := self.partner_id._get_participant_info(edi_identification)) is not None
+            and (is_company_on_peppol := self.partner_id._check_peppol_participant_exists(participant_info, edi_identification))
+        ):
+            error_msg = _(
+                "A participant with these details has already been registered on the network. "
+                "If you have previously registered to an alternative Peppol service, please deregister from that service, "
+                "or request a migration key before trying again. "
+            )
+            if (external_provider := self.partner_id._get_peppol_provider(participant_info)) and "Odoo" not in external_provider:
+                error_msg += _("The Peppol service that is used is %s.", external_provider)
+        return is_company_on_peppol, external_provider, error_msg
```

Comment:
Feel free to ignore if you don't agree:
What about returning a dictionary? It's far more flexible if you have to change things in stable one day :roll_eyes: